File: LeetCode/Study Plan/Practice Problems/Medium/Jump Game II/Java/jumpGameTwo.java
class Solution {
    public int jump(int[] nums) {
        int curr = 0, minJumps = 0, farthestJump = 0;
        int size = nums.length;

        for (int i=0; i<size-1; i++) {
            farthestJump = Math.max(farthestJump, i+nums[i]);
            if (i == curr) {
                minJumps++;
                curr = farthestJump;
            }
        }
        
        return minJumps;
    }
}

File: LeetCode/Medium/Jump Game II/Go/jumpGameTwo.go
package main

import "fmt"

func jump(nums []int) int {
	size := len(nums)
	minJumps := 0
	curr, farthest := 0, 0

	for i := 0; i < size-1; i++ {
		farthest = max(farthest, i+nums[i])
		if i == curr {
			minJumps += 1
			curr = farthest
		}
	}

	return minJumps
}

func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}

func main() {
	fmt.Println(jump([]int{2, 3, 1, 1, 4}))
}
